devito/visitors.py

```diff
@@ -413,8 +413,7 @@ def visit_Iteration(self, o, subs={}, offsets=defaultdict(set)):
             init = []
             for i, off in enumerate(filter_ordered(offsets[o.dim])):
                 vname = "%s%d" % (o.dim.name, i)
-                # Add modulo to value so that t-modulo is always positive
-                value = o.dim.parent + off + o.dim.modulo
+                value = o.dim.parent + off
```

Comment:
why did you remove that? The `+modulo` is making sure indices are always cyclic. without it the first 3 time steps are not which is maybe why this one change broke the adjoint test.